server/datastore/mysql/scim_test.go

```diff
@@ -1051,17 +1051,17 @@ func testListScimGroups(t *testing.T, ds *Datastore) {
 
 	// Test 2: Pagination - first page with 2 items
 	page1Groups, totalPage1, err := ds.ListScimGroups(context.Background(), fleet.ScimListOptions{
-		Page:    1,
-		PerPage: 2,
+		StartIndex: 1,
+		PerPage:    2,
 	})
 	require.Nil(t, err)
 	assert.Equal(t, 2, len(page1Groups))
 	assert.GreaterOrEqual(t, totalPage1, uint(3)) // Total should be at least 3
 
 	// Test 3: Pagination - second page with 2 items
 	page2Groups, totalPage2, err := ds.ListScimGroups(context.Background(), fleet.ScimListOptions{
-		Page:    2,
-		PerPage: 2,
+		StartIndex: 3, // StartIndex is 1-based, so for the second page with 2 items per page, we start at index 3
```

Comment:
I'm confused. If `StartIndex: 1` yields the first page, wouldn't `StartIndex: 3` yield the third page rather than the second page? Or is this referring to the index of the list element rather than the index of the page?